### ISSUE TITLE
issue: React crashes on Firefox/Safari when using React Hook Form with <output /> 

### ISSUE THREAD

**Issue by ekaradon** [react-hook-form/react-hook-form#10787]
### Version Number

7.45.4

### Codesandbox/Expo snack

https://codesandbox.io/s/morning-leftpad-mqzq7j

### Steps to reproduce

1. Use Firefox OR Safari, don't use Chromium or versions based on Chromium
2. Try to use the working custom version (without react hook form in the sandbox), type some text, typed text should appear next to the input, try to click on reset, it works the input is now empty
3. Try to use the working React Hook Form version (using the Div), type some text, typed text should appear next to the input, try to click on reset, it works the input is now empty
4. Try to use the not working React Hook Form version (using the Output), type some text, typed text should appear next to the input, try to click on reset, it crashes

![image](https://github.com/react-hook-form/react-hook-form/assets/9439390/cf9c3770-024d-44d1-9c5e-f4a4ae94f042)

I'm not really totally sure on how the choice to use `output` instead of `div` can make it crash? So maybe there is an issue upstream with React, but I tried to reproduce with a naive version of a form simulating RHF implementation and it works everywhere. So for now I'm only reporting it here. I hope it helps! 

### Expected behaviour

It should handle the output and not throw. 

### What browsers are you seeing the problem on?

Firefox, Safari

### Relevant log output

```shell
Node.removeChild: The node to be removed is not a child of this node
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
- fix your codesandbox
- this is most likely not an issue with RHF

**Comment by Moshyfawn** [comment]
The issue isn't RHF specific. You can read [this StackOverflow post](https://stackoverflow.com/a/64768491) that tries to explain the issue React's having with "syncing" the DOM with its virtual DOM.

TL;DR you need to wrap the conditionally rendered part of your tree in a "static" element.

**Comment by ekaradon** [comment]
I fail to see how this can relate to the usage of `output` instead of a `div` to be honest? I'm not removing any root parent here?

```tsx
    <form onSubmit={form.handleSubmit(onSubmit)}>
      <input {...form.register("name")} type="text" autoComplete="off" />
      <button type="submit">Reset!</button>
      <output>{isDirty && <span>{form.watch("name")}</span>}</output>
    </form>
```

**Comment by Moshyfawn** [comment]
Here's [an example codesandbox](https://codesandbox.io/s/upbeat-brook-vkfqzm?file=/src/RHKForm.tsx):

1. Click "Reset!" next to the "Not Working React Hook Form (Output)" input
2. Click it again
3. See the `Node.removeChild` error

Notice there're no RHF methods used in the `RHKForm` component.

Now try wrapping the conditional content inside the output element into a `div`:

```diff
- <output>{bool && <span>{name}</span>}</output>
+ <output>
+   <div>{bool && <span>{name}</span>}</div>
+ </output>
```

Notice the `Node.removeChild` isn't thrown anymore

**Comment by ekaradon** [comment]
Oh, indeed, unrelated to RHF indeed. I still don't understand how `output` instead of `div` changes this behavior but it looks like a React issue then? Thanks @Moshyfawn then! Will keep this closed then!

**Comment by Moshyfawn** [comment]
Like mentioned in [the StackOverflow post I've previously linked](https://stackoverflow.com/a/64768491), it does seem like a React issue. My guess would be that it's got to do with how the `output`'s inner content updated by the browser when the related `input` value changes vs React diff'ing it's virtual DOM. But I can't answer for sure.

No problem!
